pandas/core/indexes/base.py

```diff
@@ -3238,7 +3240,15 @@ def _convert_arr_indexer(self, keyarr):
         -------
         converted_keyarr : array-like
         """
-        keyarr = com.asarray_tuplesafe(keyarr)
+
+        if (
+            isinstance(keyarr, list)
+            and all(isinstance(k, str) for k in keyarr)
+            and needs_i8_conversion(self.dtype)
+        ):
+            keyarr = constrn.array(keyarr, self.dtype)
```

Comment:
we can end up with a PandasArray here; which we probably want to unwrap?

Reply:
@jbrockmendel 
i changed code. but yes "_convert_arr_indexer" can return either PandasArray or NumpyArray.